src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -425,12 +425,8 @@ public async Task GetStringAsync_CanBeCanceled()
                 {
                     await server.AcceptConnectionAsync(async connection =>
                     {
+                        await Task.Delay(10);
```

Comment:
If the current code is subject to a race condition, so is using a 10ms timeout.

The right fix here is to change the server to read the headers and then cancel.